launch_xml/launch_xml/entity.py

```diff
@@ -53,8 +55,23 @@ def parent(self) -> Optional['Entity']:
     @property
     def children(self) -> List['Entity']:
         """Get the Entity's children."""
+        self.__read_children = {item.tag for item in self.__xml_element}
         return [Entity(item) for item in self.__xml_element]
 
+    def assert_entity_complely_parsed(self):
+        unparsed_nested_tags = {item.tag for item in self.__xml_element} - self.__read_children
+        if unparsed_nested_tags:
+            raise ValueError(
+                f'Found the following nested tags of `{self.__xml_element.tag}` that were not '
+                f'parsed: {unparsed_nested_tags}'
```

Comment:
Tiny nitpick: from a user perspective, I don't know if "parsed" is the right word. I suggest rewording to something like,  `"Unexpected nested tag(s) found in '{self.__xml_element.tag}': {unparsed_nested_tags}"`

Reply:
Improved this one and similar error messages in https://github.com/ros2/launch/pull/468/commits/6ad4a3a47e0b50ed9d7ea581a159ccbffec821df.